src/translate.py

```diff
@@ -3397,12 +3509,17 @@ def process_instr(instr: Instruction) -> None:
                 switch_value = args.reg(0)
 
         elif mnemonic in CASES_FN_CALL:
+            is_known_void = False
             if mnemonic == "jal":
                 fn_target = args.imm(0)
                 if isinstance(fn_target, AddressOf) and isinstance(
                     fn_target.expr, GlobalSymbol
                 ):
-                    pass
+                    typemap = stack_info.global_info.typemap
+                    if typemap:
+                        c_fn = typemap.functions.get(fn_target.expr.symbol_name)
+                        if c_fn and c_fn.ret_type is None:
+                            is_known_void = True
```

Comment:
Would it be better to go off of `GlobalInfo.address_of_gsym(...)` here? 
The additional type info would not always be available, but if you were decomping multiple functions at once you could incorporate the inferred type signature from earlier passes?

Reply:
This is intentional, see the updated comment near the function call handling. We used to get some ERROR(read from unset register v0)'s before due to this being out of sync with output_regs_for_instr (which is used to construct phi nodes).